components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeConfiguration.java

```diff
@@ -17,23 +17,59 @@
 package org.apache.camel.component.xchange;
 
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
 import org.apache.camel.spi.Metadata;
 import org.apache.camel.spi.UriParam;
 import org.apache.camel.spi.UriParams;
 import org.apache.camel.spi.UriPath;
 import org.apache.camel.util.ObjectHelper;
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.currency.Currency;
 import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.utils.Assert;
 
 @UriParams
 public class XChangeConfiguration {
 
+    static final String SERVICE_MARKETDATA = "marketdata";
+    static final String SERVICE_METADATA = "metadata";
+    
+    static List<String> supportedServices = Arrays.asList(SERVICE_MARKETDATA, SERVICE_METADATA);
+    
+    static final String METHOD_CURRENCIES = "currencies";
+    static final String METHOD_CURRENCY_METADATA = "currencyMetaData";
+    static final String METHOD_CURRENCY_PAIRS = "currencyPairs";
+    static final String METHOD_CURRENCY_PAIR_METADATA = "currencyPairMetaData";
+    static final String METHOD_TICKER = "ticker";
+
+    static List<String> supportedMethods = Arrays.asList(
+            METHOD_CURRENCIES, METHOD_CURRENCY_METADATA, METHOD_CURRENCY_PAIRS, METHOD_CURRENCY_PAIR_METADATA, METHOD_TICKER);
+    
+    static final String HEADER_CURRENCY = "Currency";
+    static final String HEADER_CURRENCY_PAIR = "CurrencyPair";
+    
     @UriPath(description = "The exchange to connect to") @Metadata(required = "true")
     private String name;
+    @UriParam(description = "The service to call") @Metadata(required = "true")
```

Comment:
You can add enum="a,b,c" with the valid values these can have, then its part of the meta-data and tooling would be able to know - we do this for other components.